<a href="https://colab.research.google.com/github/vishnuwadkar/Data-Science/blob/main/Task_2_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Table Booking and Online Delivery**

In [ ]:
import pandas as pd

### Determining the Percentage of Restaurants that Offer Table Booking and Online Delivery

In [ ]:
data = pd.read_csv('Dataset .csv')

##### Calculating the restaurants that offer table booking

In [ ]:
data.columns

Index(['Restaurant ID', 'Restaurant Name', 'Country Code', 'City', 'Address',
       'Locality', 'Locality Verbose', 'Longitude', 'Latitude', 'Cuisines',
       'Average Cost for two', 'Currency', 'Has Table booking',
       'Has Online delivery', 'Is delivering now', 'Switch to order menu',
       'Price range', 'Aggregate rating', 'Rating color', 'Rating text',
       'Votes'],
      dtype='object')

In [ ]:
table_booking_count = data['Has Table booking'].value_counts()
print("The number of restaurants offering table booking are: ")
print(table_booking_count)

The number of restaurants offering table booking are: 
Has Table booking
No     8393
Yes    1158
Name: count, dtype: int64


###### Calculating percentage

In [ ]:
table_booking_perc = data['Has Table booking'].value_counts(normalize=True) * 100  #calculating the percentage through frequency of yes and no
print("The number of restaurants offering table booking are: ")
print(table_booking_perc)

The number of restaurants offering table booking are: 
Has Table booking
No     87.875615
Yes    12.124385
Name: proportion, dtype: float64


#### Therefore, approximately 12 percentage of restaurants offer table booking

### Comparing the average rating of restaurants with and without table booking

###### Let's first filter the restaurant with table booking

In [ ]:
with_table_booking = data[data['Has Table booking'] == 'Yes']

###### Then we will filter the restaurant without table booking

In [ ]:
without_table_booking = data[data['Has Table booking'] == 'No']

##### Calculating the average for each group

In [ ]:
avg_with_table_booking = with_table_booking['Aggregate rating'].mean()
avg_without_table_booking = without_table_booking['Aggregate rating'].mean()

In [ ]:
print(f"Average rating for restaurants with table booking: {avg_with_table_booking}")
print(f"Average rating for restaurants without table booking: {avg_without_table_booking}")

Average rating for restaurants with table booking: 3.4419689119170984
Average rating for restaurants without table booking: 2.559358989634219


#### Thus from the results, we can conclude that restaurants that have facility of table booking have higher ratings than the restaurants without table booking.

### Analyzing availibility of online delivery among restaurants with different prices

In [ ]:
online_delivery = pd.crosstab(data['Has Online delivery'], data['Price range'], normalize = 'index')*100
# here we did the cross tabulation of 'has online delivery ' column and 'price range' column
print("Availbility of online delivery among restaurants with different prices: ")
print(online_delivery)

Availbility of online delivery among restaurants with different prices: 
Price range                  1          2          3         4
Has Online delivery                                           
No                   52.718310  25.732394  14.042254  7.507042
Yes                  28.600571  52.468380  16.768666  2.162383


#### from this we can conclude the following

1.   For price range 1: 28.6% of the restaurants offers online delivery
2.   For price range 2: 52.47% of the restaurants offers online delivery
3.   For price range 3: 16.77% of the restaurants offers online delivery
4.   For price range 4: 2.16% of the restaurants offers online delivery



### Thus we can understand that availability of online delivery varies according to price ranges. Price Range 2 has the highest percentage of restaurants offering online delivery, while price eange 1 and price range 4 have the lowest percentages.